### ISSUE TITLE
Set_default_device() doesn't work for classes extending from Calabash::ABase

### ISSUE THREAD

**Issue by csytsma** [calabash/calabash-android#573]
Using set_default_device in a Step file works as expected.

If using the PageObject model, and extending from Calash::ABase, any screen interactions don't recognize the @default_device and fail. When debugging operations.rb, the default_device is not set.  If debugging operations.rb from a Step file, it does reflect the default_device being set.

As an example, in my Step file, if I call a screen interaction after setting set_default_device to Device2, it works.  Then on the next line of my Step definition I tell the screen to perform an action, it fails and says there are 2 devices attached.

```ruby
Then(/^I change device to "(.*?)"$/) do |device|
  info("[global_steps] Current device is #{default_device.serial}")
  set_default_device(device)
  info("[global_steps] Set device to #{default_device.serial}")
  start_test_server_in_background
  touch("* id:'login_name'")    # this will correctly interact with device2
  @screen.login_screen.touchLoginName  # this will fail
end
```

**Comment by TobiasRoikjer** [comment]
Looking at the current code:
```ruby
   def default_device
      unless @default_device
        @default_device = Device.new(self, ENV["ADB_DEVICE_ARG"], ENV["TEST_SERVER_PORT"], ENV["APP_PATH"], ENV["TEST_APP_PATH"])
      end
      @default_device
    end

    def set_default_device(device)
      @default_device = device
    end
```

It is clear why the default device does not work as expected. We should be using a class-variable instead.

**Comment by TobiasRoikjer** [comment]
Fixed in calabash 0.5.9

**Comment by csytsma** [comment]
Great, thanks Tobias!
